app/models/offer.rb

```diff
@@ -191,6 +191,12 @@ def set_initial_state
       offer.inactive_at = Time.now
     end
 
+    after_transition on: :cancel do |offer, _transition|
+      if User.current_user == offer.created_by
+        offer.expire_shareable_resource
+      end
```

Comment:
why the user condition 🤔  ?

Reply:
When Donor cancels offer, we want to auto stop sharing that offer and hence condition added.

Reply:
My point is that I think we should cancel regardless of who cancels the offer

Reply:
Also, it is dangerous to rely on a state (current_user) which only exists in the context of a request.